### PR TITLE
Update System.CommandLine from 2.0.0-beta3 to beta4

### PR DESCRIPTION
**Pull request by am11**

### CODE REVIEW

File: eng/Versions.props [hunk 1]
```diff
@@ -137,7 +137,7 @@
     <optimizationPGOCoreCLRVersion>1.0.0-prerelease.22355.3</optimizationPGOCoreCLRVersion>
     <!-- Not auto-updated. -->
     <MicrosoftDiaSymReaderNativeVersion>16.9.0-beta1.21055.5</MicrosoftDiaSymReaderNativeVersion>
-    <SystemCommandLineVersion>2.0.0-beta3.22114.1</SystemCommandLineVersion>
+    <SystemCommandLineVersion>2.0.0-beta4.22355.1</SystemCommandLineVersion>
```
Comment (jkotas) [inline eng/Versions.props:140]: The version is also listed here: https://github.com/dotnet/runtime/blob/main/eng/Version.Details.xml#L51-L54

**Comment by jkotas** [comment]
@dotnet/runtime-infrastructure Why is this not getting updated via automated code-flow?

File: eng/Versions.props (hunk 1 above)
Comment (am11) [reply eng/Versions.props:140]: Repo build succeeded for me locally without updating Details.xml, perhaps we can add a target to ensure both files are in sync?

Captured commit hash on Ubuntu with:

```sh
$ apt install -y exiftool
$ exiftool ~/.nuget/packages/system.commandline/2.0.0-beta4.22355.1/lib/net6.0/System.CommandLine.dll \
     | grep "Product Version  "
```

Comment (jkotas) [reply eng/Versions.props:140]: This should be getting update automatically via codeflow that keeps it in sync. dotnet-maestro bot should be automatically submitting a PR like this https://github.com/dotnet/runtime/pull/71900/files to update the system.commandline reference. It is not happening for some reason.

File: src/coreclr/tools/InjectResource/Program.cs [hunk 2]
```diff
@@ -1,12 +1,12 @@
-﻿// Licensed to the .NET Foundation under one or more agreements.
+// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 using System.CommandLine;
 using System.IO;
 
-var binOption = new Option<FileInfo?>(
+var binOption = new Option<FileInfo>(
```
Comment (am11) [inline src/coreclr/tools/InjectResource/Program.cs:6]: @jkotas, these were failing with nullability warnings on some legs (with `./build.sh mono` but not with `./build.sh clr`).

**Comment by am11** [comment]
osx-x64 failure is #71890.

**Review by jkotas — APPROVED** [review]
Thanks

**Pull request merged**